cmd/deploy/deploy.go

```diff
@@ -101,15 +97,12 @@ func deployFn(cmd *cobra.Command, args []string) error {
 	if _, err := exec.LookPath("kubectl"); err != nil {
 		return fmt.Errorf("install kubectl before running deploy: %v", err)
 	}
-	kubecfg, err := cmd.Flags().GetString("kubecfg")
-	if err != nil {
-		return err
-	}
 	d, err := newDeployment(args[0], false)
 	if err != nil {
 		return err
 	}
-	if err := d.Deploy(cmd.Context(), kubecfg); err != nil {
+	d.Progress = viper.GetBool("progress")
+	if err := d.Deploy(cmd.Context(), viper.GetString("kubecfg")); err != nil {
```

Comment:
What happens if the GetString returns ""?  It would be nice if this (and possibly other) values could be sanity checked prior to running these functions as then you don't have to worry about checking the values.

Reply:
this is the same behavior as the current implementation, if the flag was the empty string then the function call would handle it as before. in the case of kubecfg the empty string is completely valid as an argument if the user wants to fallback to the in cluster config.